### PR TITLE
added watch later functionality, refractored some old code

### PR DESCRIPTION
**Pull request by JantuDeb**
# Watch Later.
- add to watch later.
- remove from watch later.
- refactored some old code.
- removed unnecessary `logs`

image

### CODE REVIEW

File: src/components/shared/video-card/VideoCardActionMenu.jsx [hunk 1]
```diff
@@ -3,66 +3,73 @@ import { MdThumbDown, MdThumbUp, MdVideoLibrary } from "react-icons/md";
...
+    isLiked ? removeFromLikes(videoId) : addToLikes(videoId);
 
-  const removeVideoHandler = (videoId, playlistId) => {
-    if (videoId || playlistId) removeFromPlaylist({ playlistId, videoId });
-  };
+  const removeVideoHandler = (videoId, playlistId) =>
+    playlistId && removeFromPlaylist({ playlistId, videoId });
+
+  const watchLaterClickHandler = () =>
+    isInWatchLater ? removeFromWatchLater(videoId) : addToWatchLater(videoId);
```
Comment (Bhakti199) [inline src/components/shared/video-card/VideoCardActionMenu.jsx:36]: defining separate functions for repeating task is an good idea, liked it.

File: src/context/watch-later/WatchLaterVideoContext.jsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
+import React, { createContext, useContext, useEffect, useReducer } from "react";
+import { axioxPrivate } from "../../utils/axios-instance";
+import {
+  ADD_TO_WATCH_LATER,
+  GET_WATCH_LATER_VIDEOS,
+  REMOVE_FROM_WATCH_LATER,
+  watchLaterReducer,
+} from "./watch-later-reducer";
```
Comment (Bhakti199) [inline src/context/watch-later/WatchLaterVideoContext.jsx:8]: liked this to avoid typo error.

File: src/index.js [hunk 3]
```diff
@@ -7,6 +7,7 @@ import { SidebarProvider } from "./context/sidebar/SidebarContext";
 import { VideoProvider } from "./context/videos/VideoContext";
 import { PlaylistProvidder } from "./context/playlist/PlaylistContext";
 import { LikedVideoProvider } from "./context/liked-videos/LikedVideoContext";
+import { WatchLaterVideoProvider } from "./context/watch-later/WatchLaterVideoContext";
```
Comment (Bhakti199) [inline src/index.js:10]: you may export this from single file.

**Review by Bhakti199 — APPROVED** [review]
- This code looks fine to me.
- Well structured and written.
- Detailing of every feature and edge cases are properly handled.
- Liked feature of editing playlist.
- Proper defining of functions is done for playlist management.
- Everything looks cool.
- This PR is good to merge.

**Review by Manju-sree — APPROVED** [review]
Everything looks good in this PR. Very well written clean code and best naming convention given to variables and function names and good to see that you have also refactored code by removing unwanted logs.

File: src/App.js [hunk 4]
```diff
@@ -5,14 +5,14 @@ import Layout from "./pages/Layout";
...
 function App() {
   return (
       <Routes>
         <Route path="/" element={ <Layout><Home /></Layout>} />
         <Route path="/playlist" element={ <Layout><PlayList /></Layout>} />
-        {/* <Route path="/watch-later" element={ <Layout><WatchLater /></Layout>} /> */}
+        <Route path="/watch-later" element={ <Layout><WatchLater /></Layout>} />
         <Route path="/liked-videos" element={ <Layout><LikedVideo /></Layout>} />
         {/* <Route path="/history" element={ <Layout><History /></Layout>} /> */}
         <Route path="/video" element={<VideoDetails />} />
```
Comment (rkapoor10) [inline src/App.js:18]: you can add `<Route path="*" element={<Error/>}/>` now

**Review by rkapoor10 — APPROVED** [review]
The overall code looks well structured and is good to merge.

File: src/App.js (hunk 4 above)
Comment (JantuDeb) [reply src/App.js:18]: Will add it thanks

File: src/components/shared/video-card/VideoCardActionMenu.jsx (hunk 1 above)
Comment (JantuDeb) [reply src/components/shared/video-card/VideoCardActionMenu.jsx:36]: thanks

**Pull request merged**